### PR TITLE
feat: Optional negative points for answering a question incorrectly

### PR DESCRIPTION
**Pull request by govind007**

### CODE REVIEW

File: src/routes/PlayQuiz/QuestionsList.tsx [hunk 1]
```diff
@@ -65,7 +65,9 @@ export default function QuestionsList({
                   onClick={() => showQuestion(question.questionId, category.categoryId)}
                   key={question.questionId}>
                   <Group position="apart" style={{ width: '100%' }}>
-                    {isQuestionPointsHidden && !attemptedQuestionIds.includes(question.questionId) ? (
+                    {isQuestionPointsHidden &&
+                    !attemptedQuestionIds.includes(question.questionId) &&
+                    selectedQuestion?.questionId !== question.questionId ? (
```
Comment (rajatkantinandi) [inline src/routes/PlayQuiz/QuestionsList.tsx]: Why this change in this PR?

File: src/routes/PlayQuiz/index.tsx [hunk 2]
```diff
@@ -138,7 +138,9 @@ export default function PlayQuiz({ gameId }) {
 
       if (currentTeamIndex >= 0) {
         const currentTeam = clonedTeams[currentTeamIndex];
-        currentTeam.score = (currentTeam.score || 0) + (isCorrect ? parseInt(selectedQuestion.points.toString()) : 0);
+        currentTeam.score =
+          (currentTeam.score || 0) +
+          (isCorrect ? parseInt(selectedQuestion.points.toString()) : negativePointsForIncorrect);
```
Comment (rajatkantinandi) [inline src/routes/PlayQuiz/index.tsx]: Why are we doing toString and then parseInt again?

File: src/routes/ConfigureGame/index.tsx [hunk 3]
```diff
@@ -204,6 +212,41 @@ export default function ConfigureGame({ quizId, userName = 'guest' }) {
...
+            <Checkbox
+              radius="xl"
+              size="md"
+              mb="xs"
+              ml="md"
+              label="Allow negative response for incorrect response"
+              checked={negativePointsForIncorrect !== 0}
+              onChange={() => {
+                if (negativePointsForIncorrect === 0) {
+                  setValue('negativePointsForIncorrect', -1);
```
Comment (rajatkantinandi) [inline src/routes/ConfigureGame/index.tsx]: let's make default = 50% of correct point and also make this percentage rather than a fixed value since different questions can have different points. So, if one question has 100 points then -10 for incorrect and one question with 1000 points still having -10 for incorrect does not make sense.
Let user allow the percentage and then we calculate the negative point based on percentage.

Comment (rajatkantinandi) [reply src/routes/ConfigureGame/index.tsx]: We can even restrict it to some fixed values if we do a dropdown instead "50%", "25%", "75%" etc.

Comment (rajatkantinandi) [reply src/routes/ConfigureGame/index.tsx]: Also after calculating score make sure to round the score to 2 decimals.

Comment (govind007) [reply src/routes/ConfigureGame/index.tsx]: percentage will make the logic more complicate and it will not be user friendly. Later we can allow user to upload a spreadsheet which has different negative points for different questions (it can have percentage value also) and it will be a paid feature.
We shouldn't provide everything for free to users

Comment (govind007) [reply src/routes/ConfigureGame/index.tsx]: I was thinking to make `negative points` feature as paid, as it is not a required feature and users who are using our app for professional purpose will use this feature

File: src/routes/PlayQuiz/QuestionsList.tsx (hunk 1 above)
Comment (govind007) [reply src/routes/PlayQuiz/QuestionsList.tsx]: not related to this feature. this is a bug fixing

File: src/routes/PlayQuiz/index.tsx (hunk 2 above)
Comment (govind007) [reply src/routes/PlayQuiz/index.tsx]: removed

File: src/routes/ConfigureGame/index.tsx (hunk 3 above)
Comment (rajatkantinandi) [reply src/routes/ConfigureGame/index.tsx]: Yeah, we can make it paid feature later. Initially every feature will be free until we get enough users. We can roll out a subscription model later.

But, percentage will not be confusing to users. It is usually happens that way only. Uploading csv for different negative points will be much more painful rather than just adding a percentage that works for every question automatically.

Comment (rajatkantinandi) [reply src/routes/ConfigureGame/index.tsx]: For simplicity we can just let them choose some default options only: "1/2", "1/4", "1/3". So, instead of input field we show a select with these 3 options.

Comment (rajatkantinandi) [reply src/routes/ConfigureGame/index.tsx]: Default can be 1/4 or 1/2.

Comment (rajatkantinandi) [reply src/routes/ConfigureGame/index.tsx]: Below we show an example: "If question point is 100 then upon answering incorrectly, we deduct 50 points" when they select 1/2 for example.

Comment (rajatkantinandi) [reply src/routes/ConfigureGame/index.tsx]: So, it will be more clear

File: src/routes/PlayQuiz/QuestionsList.tsx (hunk 1 above)
Comment (govind007) [reply src/routes/PlayQuiz/QuestionsList.tsx]: created separate PR for this bug
https://github.com/rajatkantinandi/quizmaster/pull/72

File: src/components/QuestionPlay/index.tsx [hunk 4]
```diff
@@ -69,9 +71,20 @@ export default function QuestionPlay({
...
+            <Text weight="bold" component="span" size="sm">
+              Points: {points}
+            </Text>
+          ) : (
+            <Group spacing="xl">
+              <Badge color="green" variant="filled">
+                Correct: {points} points
+              </Badge>
+              <Badge color="red" variant="filled">
+                Incorrect: {points * negativePointsMultiplier} points
```
Comment (rajatkantinandi) [inline src/components/QuestionPlay/index.tsx]: negativePointsMultiplier = 0.25, 0.5, 0.33

```js
(points * negativePointsMultiplier).toFixed(2);
```
